packages/strapi/lib/middlewares/public/index.js

```diff
@@ -28,19 +32,34 @@ module.exports = strapi => {
           strapi.config.paths.static
       );
 
-      // Serve /public index page.
-      strapi.router.get(
-        '/',
-        async (ctx, next) => {
-          ctx.url = path.basename(`${ctx.url}/index.html`);
-          await next();
-        },
-        koaStatic(staticDir, {
-          maxage: maxAge,
-          defer: true,
-        })
+      // Open the file.
+      const filename =
+        strapi.config.environment === 'development' ? 'index' : 'production';
+      const index = fs.readFileSync(
+        path.join(staticDir, `${filename}.html`),
+        'utf8'
       );
 
+      const serveDynamicFiles = async ctx => {
+        ctx.url = path.basename(`${ctx.url}/${filename}.html`);
+
+        const isInitialised = await utils.isInitialised(strapi);
```

Comment:
you should do this outside the middleware to avoid recomputing or memoize the isInitialized function